testsupport/wait/host.go

```diff
@@ -769,6 +769,57 @@ func (a *HostAwaitility) GetToolchainConfig() *toolchainv1alpha1.ToolchainConfig
 	return config
 }
 
+// ToolchainConfigWaitCriterion a function to check that an ToolchainConfig has the expected condition
+type ToolchainConfigWaitCriterion func(*HostAwaitility, *toolchainv1alpha1.ToolchainConfig) bool
+
+func UntilToolchainConfigHasSyncedStatus(expectedCondition toolchainv1alpha1.Condition) ToolchainConfigWaitCriterion {
+	return func(a *HostAwaitility, toolchainConfig *toolchainv1alpha1.ToolchainConfig) bool {
+		return hasToolchainConfigSyncedStatus(a.T, expectedCondition, toolchainConfig)
+	}
+}
+
+func hasToolchainConfigSyncedStatus(t *testing.T, expectedCondition toolchainv1alpha1.Condition, toolchainConfig *toolchainv1alpha1.ToolchainConfig) bool {
+	if test.ContainsCondition(toolchainConfig.Status.Conditions, expectedCondition) {
+		t.Logf("status conditions match in ToolchainConfig")
+		return true
+	}
+	t.Logf("waiting for status condition of ToolchainConfig. Actual: '%+v'; Expected: '%+v'", toolchainConfig.Status.Conditions, expectedCondition)
+	return false
+}
```

Comment:
couldn't be the content of `hasToolchainConfigSyncedStatus` part of `UntilToolchainConfigHasSyncedStatus`? Or is there any reason for keeping it in separated function?

Reply:
🤷‍♂️  I noticed it was like that for a couple of the other functions eg. https://github.com/codeready-toolchain/toolchain-e2e/blob/03450280870985ff393b5cf445d14088d6dba2e2/testsupport/wait/member.go#L817 So I thought it was done for some reason.

I'll just make it a single function: ef269aeb7f7686ab2cc1992474c672dea13271a9

Reply:
`hasMemberStatusUsageSet` is done for a reason - it's used at two places - in `wait/host.go` & `wait/member.go` files

Reply:
ah, ok 🙂